### PR TITLE
feat: add v2 addresses endpoints

### PR DESCRIPTION
**Pull request by rafaelcr**
Create 2 new address transaction endpoints:
* `/extended/v2/addresses/:address/transactions` returning a list of transactions for an address including transfer summaries ([example](https://stacks-blockchain-p6n737eqp-blockstack.vercel.app/#tag/Transactions/operation/get_address_transactions))
* `/extended/v2/addresses/:address/transactions/:tx_id/events` returning a list of transfers for a specific transaction pertaining to 1 address ([example](https://stacks-blockchain-p6n737eqp-blockstack.vercel.app/#tag/Transactions/operation/get_address_transaction_transfers))

This also deprecates the old `/transactions_with_transfers` endpoint

### CODE REVIEW

**Comment by rafaelcr** [comment]
Will add expanded counts for mint, burn and transfer for the Explorer

File: src/tests/address-tests.ts [hunk 1]
```diff
@@ -453,6 +456,108 @@ describe('address tests', () => {
...
+
+    const v2Fetch2 = await supertest(api.server).get(
+      `/extended/v2/addresses/${testAddr2}/transactions/${v2Fetch1Json.results[0].tx.tx_id}/transfers`
+    );
+    expect(v2Fetch1.status).toBe(200);
+    expect(v2Fetch1.type).toBe('application/json');
+    expect(JSON.parse(v2Fetch2.text)).toStrictEqual({
+      limit: 20,
+      offset: 0,
+      results: [
```
Comment (zone117x) [inline src/tests/address-tests.ts:493]: Would this test be more useful if it set a limit to less than total? E.g. `limit=3` then ensure the response is ordered/correct?

Comment (zone117x) [reply src/tests/address-tests.ts:493]: Then also a follow-up request for `offset=3` to make sure the rest come in?

File: src/datastore/pg-store-v2.ts [hunk 2]
```diff
@@ -269,4 +294,152 @@ export class PgStoreV2 extends BasePgStoreModule {
...
+  async getAddressTransactionEvents(
+    args: AddressTransactionParams & TransactionPaginationQueryParams
+  ): Promise<DbPaginatedResult<DbAddressTransactionEvent>> {
+    return await this.sqlTransaction(async sql => {
+      await assertAddressExists(sql, args.address);
+      await assertTxIdExists(sql, args.tx_id);
+      const limit = args.limit ?? TransactionLimitParamSchema.default;
+      const offset = args.offset ?? 0;
+
+      const eventCond = sql`
```
Comment (He1DAr) [inline src/datastore/pg-store-v2.ts:397]: are there indexes for sender and recipient?

File: src/datastore/pg-store-v2.ts [hunk 3]
```diff
@@ -269,4 +294,152 @@ export class PgStoreV2 extends BasePgStoreModule {
...
+        canonical = true
+        AND microblock_canonical = true
+        AND tx_id = ${args.tx_id}
+        AND (sender = ${args.address} OR recipient = ${args.address})
+      `;
+      const results = await sql<(DbAddressTransactionEvent & { count: number })[]>`
```
Comment (He1DAr) [inline src/datastore/pg-store-v2.ts:403]: would a materialized view be a good candidate here?

**Review by He1DAr — APPROVED** [review]
LGTM, left a couple of questions.

File: src/datastore/pg-store-v2.ts (hunk 2 above)
Comment (rafaelcr) [reply src/datastore/pg-store-v2.ts:397]: Yep, we're good 👍

**Pull request merged**